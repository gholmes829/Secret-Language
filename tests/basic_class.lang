# very basic class

class Circle {
    let pi = 3.14

    void init(num radius) {
        print('initializing')
    }

    num get_radius() {
        print('radius')
    }
}

num main() {
    print(Circle)
    let bb_circle = Circle(2)
    bb_circle.radius = 3
    print(bb_circle.radius)
    bb_circle.init(10)
    print(bb_circle)
    return 0
}

