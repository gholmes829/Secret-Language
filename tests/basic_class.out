<Internal Class, "Circle">
initializing
3.0
initializing
<Internal Instance of "Circle">
